KS-15-04-Music-Shop

DenisGubernatorov opened the pull request:
1) Интерфейс для реализации метода поиска аксессуара для инструмента 2) Класс Instrument. Поля класса название инструмента и количество на складе. Реализует интерфейс поиска аксессуара. 3) Класс Accessories. Поля класса название инструмента и количество на складе.

DenisGubernatorov committed:
KS-15-04-Music-Shop
1) Интерфейс для реализации метода поиска аксессуара для инструмента
2) Класс Instrument. Поля класса название инструмента и количество на складе. Реализует интерфейс поиска аксессуара.
3) Класс Accessories. Поля класса название инструмента и количество на складе.

ievetrovdev commented on src/main/kotlin/lesson15/lesson15_task4.kt:
```diff
@@ -1 +1,25 @@
+    )
+}
+
+class Accessories(
```
Для более корректной иерархии нужен еще некий базовый класс для товара с общими для всех категорий полями.

DenisGubernatorov replied:
Сделал

DenisGubernatorov committed:
KS-15-04-Music-Shop
 Изменение архитектуры класса.
1) Добавлен абстрактный класс Item со свойствами name и storageCount.
2) Классы Accessories и Instrument наследуют Item

ievetrovdev reviewed (approved):
Все верно, отличное решение!

ievetrovdev commented:
Только это не Задача 1 Урока 16, потому отклонено будет
